src/targets/registry.ts

```diff
@@ -408,20 +445,76 @@ export class RegistryTarget extends BaseTarget {
         this.logger.warn(
           `No files found that match "${onlyIfPresentPattern.toString()}", skipping the target.`
         );
+        this.published = true;
         return undefined;
       }
     }
 
-    const remote = this.registryConfig.registryRemote;
-    const username = await getAuthUsername(this.github);
-    remote.setAuth(username, getGithubApiToken());
-
-    await withTempDir(
-      directory =>
-        this.pushVersionToRegistry(directory, remote, version, revision),
-      true,
-      'craft-release-registry-'
+    const startBatch = RegistryTarget.lock.isLocked();
+    const result = RegistryTarget.lock.runExclusive(() =>
+      this.doPublish(version, revision)
     );
```

Comment:
Suggest flipping these two lines and possibly otherwise reformatting to make it clearer that we always `doPublish` for ourself, and only kick off publishing for everyone else if we're the first in the batch.


```suggestion
    const result = RegistryTarget.lock.runExclusive(() =>
      this.doPublish(version, revision)
    );
    const startBatch = RegistryTarget.lock.isLocked();
```

Reply:
Cannot do it the way you suggested as as soon as we run `RegistryTarget.lock.runExclusive`, `RegistryTarget.lock.isLocked()` will always return `true` 🙂

Reply:
(Discussing race condition in private Slack ...)

Reply:
This is no longer an issue with the eager `this.published = true` command.